src/gui/touchscreengui.cpp

```diff
@@ -678,12 +679,17 @@ void TouchScreenGUI::handleReleaseEvent(size_t evt_id)
 			auto *translated = new SEvent;
 			memset(translated, 0, sizeof(SEvent));
 			translated->EventType               = EET_MOUSE_INPUT_EVENT;
-			translated->MouseInput.X            = m_move_downlocation.X;
-			translated->MouseInput.Y            = m_move_downlocation.Y;
 			translated->MouseInput.Shift        = false;
 			translated->MouseInput.Control      = false;
 			translated->MouseInput.ButtonStates = 0;
 			translated->MouseInput.Event        = EMIE_LMOUSE_LEFT_UP;
+			if (m_use_crosshair || m_camera_mode == CAMERA_MODE_THIRD) {
```

Comment:
Instead of replicating the condition here (error prone) and even passing `m_camera_mode` around why not directly pass the current state of `draw_crosshair` to touchscreengui?

Reply:
Yes, `m_draw_crosshair` will be used. The calculation is moved into `game.cpp` (`touchscreengui.cpp` only receive the result).